File: service/app/src/main/java/io/miragon/blueprint/adapter/inbound/operaton/SendCancellationConfirmationWorker.java
package io.miragon.blueprint.adapter.inbound.operaton;

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker;
import dev.bpmcrafters.processengine.worker.Variable;
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks;
import io.miragon.blueprint.application.port.inbound.SendCancellationConfirmationUseCase;
import io.miragon.blueprint.domain.leasing.ApplicationId;
import org.springframework.stereotype.Component;

@Component
public class SendCancellationConfirmationWorker {

    private final SendCancellationConfirmationUseCase useCase;

    public SendCancellationConfirmationWorker(SendCancellationConfirmationUseCase useCase) {
        this.useCase = useCase;
    }

    @ProcessEngineWorker(topic = ServiceTasks.SEND_CANCELLATION_CONFIRMATION)
    public void sendCancellationConfirmation(@Variable String applicationId) {
        useCase.sendCancellationConfirmation(ApplicationId.of(applicationId));
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/adapter/inbound/operaton/SendCancellationConfirmationWorker.kt
package io.miragon.blueprint.adapter.inbound.operaton

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker
import dev.bpmcrafters.processengine.worker.Variable
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks
import io.miragon.blueprint.application.port.inbound.SendCancellationConfirmationUseCase
import io.miragon.blueprint.domain.leasing.ApplicationId
import org.springframework.stereotype.Component

@Component
class SendCancellationConfirmationWorker(
    private val useCase: SendCancellationConfirmationUseCase,
) {

    @ProcessEngineWorker(topic = ServiceTasks.SEND_CANCELLATION_CONFIRMATION)
    fun sendCancellationConfirmation(@Variable applicationId: String) {
        useCase.sendCancellationConfirmation(ApplicationId.of(applicationId))
    }
}
